src/KK.AspNetCore.EasyAuthAuthentication/Services/EasyAuthAzureAdService.cs

```diff
@@ -0,0 +1,34 @@
+namespace KK.AspNetCore.EasyAuthAuthentication.Services
+{
+    using System.Security.Claims;
+    using KK.AspNetCore.EasyAuthAuthentication.Interfaces;
+    using KK.AspNetCore.EasyAuthAuthentication.Models;
+    using Microsoft.AspNetCore.Authentication;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.Logging;
+
+    public class EasyAuthAzureAdService: EasyAuthWithHeaderService<EasyAuthAzureAdService>, IEasyAuthAuthentificationService
+    {
+        private readonly ILogger<EasyAuthAzureAdService> logger;
+
+        public EasyAuthAzureAdService(ILogger<EasyAuthAzureAdService> logger) : base(logger)
+        {
+            this.logger = logger;
+            this.defaultOptions = new ProviderOptions(typeof(EasyAuthAzureAdService).Name, "name", ClaimTypes.Role);
```

Comment:
there is also a change in the default name claim. Maybe we want role this back. Befor the result of the name was the upn. Now its the first and the last name.

Reply:
The reason is that in the sample you has the following line:

`this.HttpContext.User.Identity.Name`

And this should return the name instead of the upn.
But there is also no other property on the identity object that define a name. So we must discuss if the user of this library want a unique name or really the name of the user.